pyfr/integrators/std/controllers.py

```diff
@@ -175,7 +175,7 @@ def advance_to(self, t):
 
         while self.tcurr < t:
             # Decide on the time step
-            dt = max(min(t - self.tcurr, self._dt, self.dtmax), self.dtmin)
+            dt = max(min(t - self.tcurr, self.dt, self.dtmax), self.dtmin)
```

Comment:
I thought the new logic was supposed to take care of this?

Reply:
Just following up here.

Reply:
We will probably need to refactor the code a bit to remove that line. I will do that and test the edge cases by tomorrow.

Reply:
I just realised, we are already using `self.fact` and `self.fact_max` to restrict the allowed change in `dt`. We could remove `self.dtmin` everywhere. Now that we are hitting the target times exactly, is there any reason to keep `self.dtmin`?

Reply:
How would we prevent dt from going below a certain amount?

Reply:
I added a line to prevent `dt` from going below `self.dtmin`. 

`self.dt` is modified with line 78-80 to make sure that we do not change dt suddenly. The time that we will come across this condition is when `self.dt` is close to `self.dtmin` because the controller requires it. I have not come across this though, so I assumed it is not required.

Reply:
This literally happens all of the time: https://pyfr.discourse.group/search?expanded=true&q=Minimum%20sized%20time%20step%20rejected

Reply:
You're right. I replicated the issue in https://pyfr.discourse.group/t/minimum-sized-time-step-rejected-2d-pipe-flow/720 and could not think of a way to go around setting `self.dtmin`.